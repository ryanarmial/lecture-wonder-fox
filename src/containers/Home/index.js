import React, { Component } from 'react'
import axios from 'axios'

class Home extends Component {
  state = {
    person: {}
  }

  async componentDidMount() {
    this.setState({
      person: await this.getPerson()
    })
  }

  getPerson = () => {
    return axios
      .get('https://swapi.co/api/people/1')
      .then(({ data }) => {
        return data
      })
  }

  render() {
    return (
      <h1>HALO HAAI</h1>
    );
  }
}

export default Home;