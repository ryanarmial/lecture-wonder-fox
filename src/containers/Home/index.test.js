import React from 'react';
import Home from './index';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


describe('Testing Async <Home/>', () => {
  const wrapper = shallow(<Home />)
  it('Should render <Home/>', () => {
    expect(wrapper).toBeTruthy()
  })
  it('Should Get Person', async () => {
    const person = await wrapper.instance().getPerson()
    expect(person).toHaveProperty('name')
  })
  // it('Should Get State Person', () => {
  //   wrapper.update()
  //   const person = wrapper.state('person')
  //   expect(person).toHaveProperty('name')
  // })
  it('Test Snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
