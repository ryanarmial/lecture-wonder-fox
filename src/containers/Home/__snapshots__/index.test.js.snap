// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Async <Home/> Test Snapshot 1`] = `
<h1>
  HALO HAAI
</h1>
`;
